addons/website_event/static/src/js/website_event.js

```diff
@@ -88,9 +88,11 @@ publicWidget.registry.EventRegistrationFormInstance = publicWidget.Widget.extend
      * @override
      */
     start: function () {
-        var def = this._super.apply(this, arguments);
-        var instance = new EventRegistrationForm(this);
-        return Promise.all([def, instance.attachTo(this.$el)]);
+        if(!this.target){
+            var def = this._super.apply(this, arguments);
+            var instance = new EventRegistrationForm(this);
+            return Promise.all([def, instance.attachTo(this.$el)]);
+        }
     },
```

Comment:
You will need to install a linter and follow JS guidelines: https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines (there is a sample linter configuration in there).

Anyways: ask when you did not understand what is to be done here. What you need to do is:
- Currently, a "public widget" "EventRegistrationFormInstance" is automatically instantiated on the page as its "selector" matches an element on the page. This is the purpose of public widget which are in the registry: being automatically instantiated when their selector matches the DOM.
- The public widget actually instantiates an internal normal widget EventRegistrationForm and attaches it to the DOM. Why? For no good reason, just bad code that should be refactored.
- When entering edit mode, all public widgets are automatically destroyed, and rebuilt in edit mode if they ask to. This one does not ask to, so it is simply destroyed. When any widget is destroyed, all the widgets it created are destroyed too. So here, the internal EventRegistrationForm is destroyed when entering edit mode. When a widget is destroyed, unless it is a public widget which was automatically instantiated, it removes its main element from the DOM -> this is our bug here: we remove an element from the DOM by mistake just because a widget was attached to it then destroyed. What you need to do is: when the public widget is destroyed, its internal widget should behave like a public widget too, not destroying the DOM: https://github.com/odoo/odoo/blob/13.0/addons/web/static/src/js/public/public_widget.js#L233